client/src/main/components/SettingsDialog/SettingsDialog.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import {
+  Typography,
+  Button,
+  Dialog,
+  FormGroup,
+  Checkbox,
+  FormControlLabel,
+  createMuiTheme,
+} from '@material-ui/core';
+import { Theme } from '@material-ui/core/styles';
+import {
+  AlertFunction,
+  UpdateSettingsFunction,
+  UpdateUserOnServerFunction,
+} from '../../App';
+import { UserSettings } from '../../logic/dbTypes';
+import FontSizeSetting from './FontSizeSetting';
+import baseThemeOptions from '../../AppTheme';
+
+type SettingsDialogProps = {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  alert: AlertFunction;
+  settings: UserSettings;
+  appTheme: Theme;
+  setTheme: (updatedTheme: Theme) => void;
+  updateSettings: UpdateSettingsFunction;
+  sendUpdatedUserToServer: UpdateUserOnServerFunction;
+};
+
+/**
+ * Represents the settings dialog for a user.
+ */
+function SettingsDialog(props: SettingsDialogProps): JSX.Element {
+  const {
+    setOpen,
+    open,
+    alert,
+    sendUpdatedUserToServer,
+    settings,
+    updateSettings,
+    appTheme,
+    setTheme,
+  } = props;
+
+  /**
+   * Handles closing of the SettingsDialog.
+   */
+  function handleClose(): void {
+    setOpen(false);
+  }
+
+  /**
+   * Handles saving of the user settings.
+   */
+  function handleSave(): void {
```

Comment:
Small hangup when saving after changing the fonsize setting in the UI, the settingsdialog does not close after it is saved and another press of the save is required. See gif below

https://i.imgur.com/PcNLVjd.gifv

Reply:
Thank you for going through the effort of making a gif for this. I know exactly how to fix the problem because of that.

Reply:
This is fixed now!

Reply:
Looks great! nice work